src/coalescenceml/integrations/azure_datalake/artifact_store/azure_artifact_store.py

```diff
@@ -0,0 +1,83 @@
+from ast import Tuple
+import os
+from pathlib import Path
+from adlfs import AzureBlobFileSystem
+
+from coalescenceml.artifact_store.base_artifact_store import BaseArtifactStore, PathType
+from typing import (
+    Any,
+    Callable,
+    Iterable,
+    List,
+    Optional,
+)
+
+class AzureArtifactStore(BaseArtifactStore):
+    """ Artifact Store for Azure Datalake artifacts""" 
+
+    abfs: Optional[AzureBlobFileSystem] = None
+    account_name: str
+
+    def _ensure_filesystem_exists(self):
+        if not self.abfs:
+            self.abfs = AzureBlobFileSystem(account_name=self.account_name, anon=False)
+        return self.abfs
+
+    def open(self, name: PathType, mode: str = "rb") -> Any:
+        """ Open a file at""" 
+        self._ensure_filesystem_exists()
+        return self.abfs._open(name, mode=mode)
```

Comment:
Why the use of `_open` instead of just `open`? It seems this is a common pattern that you have across the board.

Reply:
I think some the of functions only have private method tho

Reply:
According to docs and code base, they have `open` as a function which is public. I think we should use the public methods where possible